### PR TITLE
Feat: add color to '* has already been recorded in the archive' message

### LINKED ISSUE #4913: Add some color to '* has already been recorded in the archive' message to highlight output of download entry
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I remove or skip any mandatory\* field

### Checklist

- [X] I'm requesting a feature unrelated to a specific site
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've verified that I'm running yt-dlp version **2022.09.01** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Provide a description that is worded well enough to be understood

Basically title. When downloaded playlists messages of already downloaded video are merged into one white big output message.
My suggestion to add some color like in '[download] Downloading video `i` of `n`' message or something like that.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
Well, there isn't any problems but here is verbose output.
[debug] Command-line config: ['-vU']
[debug] Encodings: locale cp1251, fs utf-8, pref cp1251, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.09.01 [5d7c7d6] (win32_exe)
[debug] Python 3.8.10 (CPython 64bit) - Windows-10-10.0.19041-SP0
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg 5.0-full_build-www.gyan.dev (setts), ffprobe 5.0-full_build-www.gyan.dev, phantomjs 2.1.1
[debug] Optional libraries: Cryptodome-3.15.0, brotli-1.0.9, certifi-2022.06.15, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.3
[debug] Proxy map: {}
[debug] Loaded 1670 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.09.01, Current version: 2022.09.01
yt-dlp is up to date (2022.09.01)
```

### PR DESCRIPTION
**Pull request by aaruni96**
### Adds color to the message '* has already been recorded in the archive'

This PR adds color to the info line if a video has already been recorded in the arvhice.

Closes #4913 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [x] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### CODE REVIEW

**Comment by Grub4K** [comment]
There are two occurences for the "recorded" message, one in `_match_entry` and one in `extract_info`.
Using f-strings for non legacy is prefered.
I am also not sure, but the `key` parameter in the `to_screen` should probably not have color, as that is part of the standardized output.

It makes sense to adjust the output to give the id and title if available, else only one of the two.

**Comment by pukkandan** [comment]
In addition, `reason` should not be colored

**Comment by aaruni96** [comment]
If there is no more work in this PR, but it takes time to merge, can you in the meantime tag this with `hacktoberfest-accepted` ?

File: yt_dlp/YoutubeDL.py [hunk 1]
```diff
@@ -1459,7 +1459,10 @@ def extract_info(self, url, download=True, ie_key=None, extra_info=None,
 
             temp_id = ie.get_temp_id(url)
             if temp_id is not None and self.in_download_archive({'id': temp_id, 'ie_key': key}):
-                self.to_screen(f'[{key}] {temp_id}: has already been recorded in the archive')
+                self.to_screen(f'[{key}] '
+                               f'{self._format_screen(temp_id, self.Styles.ID)}: '
+                               'has already been recorded in the archive'
+                               )
```
Comment (Grub4K) [inline yt_dlp/YoutubeDL.py:1465]: Maybe inline the second f-string? Also closing brackets should reside on the same line.
```suggestion
                self.to_screen(f'[{key}] {self._format_screen(temp_id, self.Styles.ID)}: '
                               'has already been recorded in the archive')
```
Maybe also prepend `entry` so its the same as the other message format? (`entry has already been recorded in the archive`)

File: yt_dlp/YoutubeDL.py [hunk 2]
```diff
@@ -1402,7 +1402,7 @@ def check_filter():
             return None
 
         if self.in_download_archive(info_dict):
-            reason = '%s has already been recorded in the archive' % video_title
+            reason = f'{self._format_screen(video_title, self.Styles.FILENAME)} has already been recorded in the archive'
```
Comment (Grub4K) [inline yt_dlp/YoutubeDL.py:1405]: Maybe not, but I would like to see both `id` and `title` if available. I'd like to know @pukkandan's opinion on this matter.

Comment (aaruni96) [reply yt_dlp/YoutubeDL.py:1405]: Sorry, I did not get notifications for the added labels.
@pukkandan I was waiting for your reply to the above comment before making any more changes

> I'd like to know @pukkandan's opinion on this matter.

Comment (pukkandan) [reply yt_dlp/YoutubeDL.py:1405]: I agree that it makes sense to show both

**Comment by johndrinkwater** [comment]
Came to report youtube playlist download mode doesn’t list video IDs of already archived links compared to regular youtube urls, and luckily found this patches that.

Was this PR waiting for further updates?

**Review by Grub4K — COMMENTED** [review]
This would be what I thought of

**Comment by pukkandan** [comment]
> Came to report youtube playlist download mode doesn’t list video IDs of already archived links compared to regular youtube urls, and luckily found this patches that.
> 
> Was this PR waiting for further updates?

Whatever your isue is, it does not eem related to this PR.

File: yt_dlp/YoutubeDL.py [hunk 3]
```diff
@@ -1351,7 +1351,8 @@ def prepare_filename(self, info_dict, dir_type='', *, outtmpl=None, warn=False):
     def _match_entry(self, info_dict, incomplete=False, silent=False):
         """ Returns None if the file should be downloaded """
 
-        video_title = info_dict.get('title', info_dict.get('id', 'entry'))
+        video_id = info_dict.get('id', 'entry')
+        video_title = info_dict.get('title', video_id)
```
[the remark this answers is not in the document]
Comment (pukkandan) [reply yt_dlp/YoutubeDL.py]: video_title is used elsewhere

[the remark this answers is not in the document]
Comment (Grub4K) [reply yt_dlp/YoutubeDL.py]: Oh my bad, so something like `title_format = ...`?

**Comment by pukkandan** [comment]
Ive pushed something. Let me know opinion.

**Comment by Grub4K** [comment]
I would have wanted a clearer separation for the title, but this is way cleaner code LGTM.

**Comment by pukkandan** [comment]
Hm.. match_filter messages are inconsistent whether it puts title in `""` or not. We can standardize it later

**Comment by Grub4K** [comment]
I would always put title in quotes, but yes, we can standardize later, maybe in conjunction with structural logging.

**Pull request merged**